src/openforms/api/serializers.py

```diff
@@ -69,6 +75,27 @@ class ValidationErrorSerializer(ExceptionSerializer):
     invalid_params = FieldValidationErrorSerializer(many=True)
 
 
+@dataclass
+@total_ordering
+class BatchItem:
+    priority: int
+    instance: Model
+    source: str  # index.field_name of the source data
+
+    @property
+    def model(self) -> Type[Model]:
+        return type(self.instance)
+
+    def __post_init__(self):
+        # high priority first, grouped by type
+        self._sorting_key = (-self.priority, self.model.__name__)
+
+    def __lt__(self, other: object) -> bool:
+        if not isinstance(other, BatchItem):  # pragma: no cover
```

Comment:
Not worth a unit test. I guess removing this check is also an option. Raising `AttributeError` when there is no `_sorting_key` works too.